Documented handling of /pub in the context of proxy (bsc#1238827) (#3779)
uyuni-project/uyuni-docs: modules/client-configuration/pages/client-proxy-script.adoc

This is done almost the same way as registering clients directly with the {productname} Server.

You create the bootstrap script in the {productname} Server container with the [command]``mgr-bootstrap`` command line tool.
[command]``mgr-bootstrap`` requires some parameters such as the fully qualified domain name (FQDN) of the proxy, activation keys, or GPG keys.
These parameters depend on your specific setup, but the FQDN of the proxy is mandatory.

The bootstrap script then deploys all necessary information to the clients.

.Procedure: Registering Clients to a Proxy with a Bootstrap Script

. Create a client activation key on the {productname} Server using the {webui}.
  For more information, see xref:client-configuration:activation-keys.adoc[].
. On the container host of the server, start a terminial in the container:

+

[source,shell]
----
mgrctl term
----

+

--
.. In the container, to create the bootstrap script, execute the [command]``mgr-bootstrap`` command:

+

[source,shell]
----
mgr-bootstrap --hostname=MLM_PROXY --activation-keys=ACTIVATION_KEY
----

+

Replace [literal]``MLM_PROXY`` with the host name of your proxy.

> If needed, use the additional command line switches to tune your bootstrap script.
>    To view available options type [command]``mgr-bootstrap --help`` from the command line.

Author: ```suggestion
   If needed, use the additional command line switches to tune your bootstrap script.
   To view available options type [command]``mgr-bootstrap --help`` from the command line.
```

>    To view available options type [command]``mgr-bootstrap --help`` from the command line.

Reviewer: I forgot to also mention that they should consider using custom name for the resulting script so all the bootstrap scripts are not mixed together.

E.g.:

Consider using [command]``--script bootstrap-MLM_PROXY.sh`` option to save bootstrap script with descriptive name.

.. OPTIONAL: Edit the resulting bootstrap script.
--

. Execute the bootstrap script.
  There are two options:

+

* Transfer it to the client and run it directly on the clients.

>   Alternatively, because the proxy forwards requests to the server, the client can download it from http://MLM_PROXY/pub/bootstrap.

Reviewer: ```suggestion
  Alternatively, because the proxy forwards requests to the server, the client can download it from http://MLM_PROXY/pub/bootstrap.sh.
```

Reviewer: Default bootstrap script name is `bootstrap.sh` (unless changed by the option mentioned above)

  Replace [literal]``MLM_PROXY`` with the host name of your proxy.

> * Execute it from the proxy with [command]``ssh``.

Reviewer 2: ```suggestion
* Execute the script from the proxy with [command]``ssh``.
```

[[script-client-proxy]]
= Register Clients  to a Proxy with a Bootstrap Script

You can register clients through the {productname} Proxy with a bootstrap script.
This is done almost the same way as registering clients directly with the {productname} Server.

You create the bootstrap script on the {productname} Server with a command line tool.
The bootstrap script then deploys all necessary information to the clients.
The bootstrap script requires some parameters such as the FQDN of the proxy, activation keys, or GPG keys.

> These parameters depend on your specific setup, but the FQDN of the proxy is mandatory.

Reviewer: This is little confusing. It's not a bootstrap script requiring that parameters, but the script generating bootstrap script.

> You create the bootstrap script on the {productname} Server with a command line tool.
> The bootstrap script then deploys all necessary information to the clients.
> The bootstrap script requires some parameters such as the FQDN of the proxy, activation keys, or GPG keys.
> These parameters depend on your specific setup, but the FQDN of the proxy is mandatory.

Reviewer: ```suggestion
You create the bootstrap script in the {productname} Server container with a command line tool [command]``mgr-bootstrap``.
The [command]``mgr-bootstrap`` script requires some parameters such as the FQDN of the proxy, activation keys, or GPG keys.
These parameters depend on your specific setup, but the FQDN of the proxy is mandatory.

The generated bootstrap script then deploys all necessary information to the clients.
```

.Procedure: Registering Clients to a Proxy with a Bootstrap Script

. Create a client activation key on the {productname} Server using the {webui}.
  For more information, see xref:client-configuration:activation-keys.adoc[].
. On the container host of the server, start a terminial in the container:

+

[source,shell]
----
mgrctl term
----

+

--
.. In the container, to create the bootstrap script, execute the [command]``mgr-bootstrap`` command:

+

> // FIXME or must we set --http-proxy here?

Reviewer: `--http-proxy` is when the client should use HTTP proxy to be able to connect to SMLM proxy. There are two different things. So it is not mandatory.

> 
> +
> 
> // FIXME or must we set --http-proxy here?

Author: ```suggestion

+

// FIXME or must we set --http-proxy here?
```
```suggestion
```

+

[source,shell]
----

> mgr-bootstrap --hostname=FQDN --activation-keys=ACTIVATION_KEY

Reviewer: ```suggestion
mgr-bootstrap --hostname=MLM_PROXY --activation-keys=ACTIVATION_KEY
```

+

> Replace [literal]``MLM_PROXY`` with the host name of your proxy.

Reviewer: Not sure how exactly do this, but I think it would improve readability if there was a new line after this line. Is it like this?

```suggestion
Replace [literal]``MLM_PROXY`` with the host name of your proxy.

+
```
```

Author: yes, but additionally removing indentation of the following line is needed...

> If needed, use the additional command line switches to tune your bootstrap script.
>    To view available options type [command]``mgr-bootstrap --help`` from the command line.

Reviewer: ```suggestion
Replace [literal]``MLM_PROXY`` with the host name of your proxy.
If needed, use the additional command line switches to tune your bootstrap script.
   To view available options type [command]``mgr-bootstrap --help`` from the command line.
```

>    With the [option]``--script`` option you save the bootstrap script with a descriptive name.

Author: ```suggestion
With the [option]``--script`` option you save the bootstrap script with a descriptive name.
```

.. OPTIONAL: Edit the resulting bootstrap script.
--

. Execute the bootstrap script.
  There are two options:

+

* Transfer the script to the client and run it directly on the clients.

>   Alternatively, because the proxy forwards requests to the server, the client can download it from http://MLM_PROXY/pub/bootstrap.sh.

Reviewer: As the example now modifies the script name, use that in this example as well.

```suggestion
  Alternatively, because the proxy forwards requests to the server, the client can download it from http://MLM_PROXY/pub/bootstrap-MLM_PROXY.sh.
```

  Replace [literal]``MLM_PROXY`` with the host name of your proxy.

* Execute the script from the proxy with [command]``ssh``.

>   Replace [systemitem]``<bootstrap>`` with the name of the bootstrap script and [systemitem]`` <client.example.com>`` with the host name of the client:

Reviewer: ```suggestion
  Replace [literal]``MLM_PROXY`` with the hostname of the proxy and [systemitem]`` <client.example.com>`` with the host name of the client:
```

+

[source,shell]
----

> cat <bootstrap> | ssh root@<client.example.com> /bin/bash

Reviewer: Let's standardize and clarify all examples:

```suggestion
cat /srv/www/htdocs/pub/bootstrap/bootstrap-MLM_PROXY.sh | ssh root@<client.example.com> /bin/bash
```
